pkgs/build-support/rust/cargo-local-registry.nix

```diff
@@ -0,0 +1,33 @@
+{ fetchurl, stdenv }:
+
+let
+  version = "0.1.4";
+  platform =
+    if stdenv.system == "x86_64-linux"
```

Comment:
There are a lot of dependencies and other stuff to manage. I don't think I know enough about the rust setup to do it myself but if someone can take it on that would be great. I was focused here on showing that cargo-local-registry is probably the way forward.

However building might not actually be necessary if we can rely on the cached deps pacakge. This package is archetecture independant so once we have them we can just build cargo-local-registry from source. I don't know if there is a policy about relying on the cache for things like this.